dev/tasks/homebrew-formulae/apache-arrow.rb

```diff
@@ -72,6 +72,7 @@ def install
     # link against system libc++ instead of llvm provided libc++
     ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
     args = %W[
+      -DARROW_ACERO=ON
```

Comment:
Do we need this?
`ARROW_DATASET=ON` must enable `ARROW_ACERO` automatically: https://github.com/apache/arrow/blob/ee42b908f5102fe8c73b5e84b58b155e49ada747/cpp/cmake_modules/DefineOptions.cmake#L308-L313

Reply:
I'm not sure what the guidelines are.  R uses Acero directly, in addition to transitively through datasets.  So if R decided to stop using datasets then it would still need the acero dependency.  However, if we want to leave it out so things are concise then it should work.  I am fine either way.

Reply:
I see. Thanks.

For `dev/tasks/homebrew-formulae/apache-arrow.rb`, this is a file for Homebrew's CI. (We should sync with https://github.com/Homebrew/homebrew-core/blob/master/Formula/apache-arrow.rb .) This is not only for R. In general, Homebrew enables all features as much as possible. So I think that we don't need to assume that we may disable only `ARROW_DATASET` in this file. (We should omit redundant `-DARROW_ACERO=ON`.)

For `dev/tasks/homebrew-formulae/autobrew/`, they are only for R. So I think that your point is valid.